tools/autograd/gen_variable_type.py

```diff
@@ -107,6 +370,22 @@ def create_variable_type(top_env, aten_declarations):
     type_declarations = top_env['type_derived_method_declarations']
     type_definitions = top_env['type_derived_method_definitions']
 
+    def save_variables(option, derivative):
```

Comment:
I'm not exactly sure what this function does, mostly because I don't think I understand the relation between declarations, options and derivatives. It would be nice to document it somewhere